### PR TITLE
#184651081 Buyer should be able to track their order status and get notified to the frontend.

### PR DESCRIPTION
**Pull request by ishimwe5555**
#### What does this PR do?
It handles the retrieval and updating of order status information, and provide this information to the frontend for display.
#### Description of Task to be completed?
The appropriate order status information should be retrieved from the database
The order status information should be sent to the frontend to be displayed to the buyer
The order status in the database should be updated
The updated order status should be sent to the frontend to be displayed to the buyer
#### How should this be manually tested?
Clone the repository and npm install to install all dependencies
Seng a GET request to retrieve the order's status via /orders/id
The order's status will be returned as a response and notification should be sent to the frontend
#### What are the relevant pivotal tracker/Trello stories?
https://www.pivotaltracker.com/story/show/184651081
#### Questions:

### CODE REVIEW

File: src/controllers/order.controller.js [hunk 1]
```diff
@@ -1,77 +1,4 @@
-import WebSocket from 'ws';
-import { orderServices, userServices } from '../services';
-
-// // Initialising web sockets
-// const wss = new WebSocket.Server({ port: 4000 });
```
Comment (JosephNgabo) [inline src/controllers/order.controller.js:5]: can you remove these commented ones?incase they aren't needed?

**Review by Kress2000 — APPROVED** [review]
remove all those un necessary comments, otherwise, looks good to me

File: src/controllers/order.controller.js (hunk 1 above)
Comment (ishimwe5555) [reply src/controllers/order.controller.js:5]: They are removed, that's why they in red

**Review by Pacific911 — APPROVED** [review]

**Review by Kress2000 — APPROVED** [review]

**Review by JosephNgabo — APPROVED** [review]

**Review by Pacific911 — CHANGES_REQUESTED** [review]

**Review by Kress2000 — APPROVED** [review]

**Review by mirindi19 — APPROVED** [review]

File: src/controllers/order.controller.js [hunk 2]
```diff
@@ -92,24 +20,43 @@ const getOrdersByUser = async (req, res) => {
...
+    return res.status(200).json({
+      code: '200',
+      order,
+    });
+  } catch (err) {
+    return res.status(500).json({
+      code: '500',
+      message: 'We have an error',
+      error: err.message,
+    });
```
Comment (princoo) [inline src/controllers/order.controller.js:43]: an error thrown by a controller is handled by the  `errorHandler`

File: src/routes/order.route.js [hunk 3]
```diff
@@ -6,15 +6,15 @@ import {
...
   getOrdersByUser,
+  getOrderByUser,
 } from '../controllers/order.controller';
 import { asyncWrapper } from '../helpers';
 
 const router = express.Router();
 
 router.get('/', isAuthenticated, getOrdersByUser);
-router.get('/sse', getOrderStatusEvents);
+router.get('/:orderId', getOrderByUser);
```
Comment (princoo) [inline src/routes/order.route.js:17]: if this controller throws error to the client , wrap your controller in `asyncWrapper`

Comment (ishimwe5555) [reply src/routes/order.route.js:17]: sure did

Comment (ishimwe5555) [reply src/routes/order.route.js:17]: sure did

**Review by AdrineUWERA — APPROVED** [review]

**Review by princoo — APPROVED** [review]

File: src/controllers/order.controller.js [hunk 4]
```diff
@@ -1,76 +1,4 @@
...
-
-wss.on('connection', (ws) => {
-  console.log('Client connected');
-
-  ws.on('message', (message) => {
-    console.log(`Received message: ${message}`);
-  });
-
-  ws.on('close', () => {
-    console.log('Client disconnected');
```
[the remark this answers is not in the document]
Comment (munezerobagira) [reply src/controllers/order.controller.js:32]: this is deleted

[the remark this answers is not in the document]
Comment (ishimwe5555) [reply src/controllers/order.controller.js:32]: Yeah it's from a deleted section.

[the remark this answers is not in the document]
Comment (ishimwe5555) [reply src/controllers/order.controller.js:32]: Yeah it's from a deleted section.

**Review by JosephNgabo — APPROVED** [review]

**Review by princoo — APPROVED** [review]

**Review by Pacific911 — APPROVED** [review]

**Pull request merged**